### PR TITLE
Logging depwarn cleanup

### PR DESCRIPTION
**Pull request by c42f**
Some cleanup mostly to address problems @stevengj found over at https://github.com/JuliaLang/julia/commit/5d77d1b8ddd4bbe340e087802bf928ac208e4229#r26334167

I also added some license headers which I realized were forgotten from the original PR.

Potentially replaces #25229

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
This is passing tests so feel free to merge when you feel it's ready!

File: base/client.jl [hunk 1]
```diff
@@ -137,6 +137,12 @@ function repl_cmd(cmd, out)
 end
 
 function display_error(io::IO, er, bt)
+    if !isempty(bt)
+        st = stacktrace(bt)
+        if !isempty(st)
+            io = redirect(io, log_error_to, st[1])
+        end
+    end
```
Comment (c42f) [inline base/client.jl]: @stevengj by the way, here's an answer to your question about `log_error_to` - for the moment I've reinstated it exactly as it was used inside `display_error`. I think it was a bit of a misfeature if used to capture REPL errors, but it's easy to preserve the behavior for now.

Comment (stevengj) [reply base/client.jl]: The point is, this will get removed when the deprecation is removed in 1.0, right?

Comment (c42f) [reply base/client.jl]: Right.  The user should get a depwarn if they ever manage to populate `log_error_to`.

**Comment by c42f** [comment]
Cheers!  This was a relatively minor cleanup so I went ahead and merged it without further ado.

**Pull request merged**